cmd/frontend/graphqlbackend/user.go

```
@@ -169,7 +170,7 @@ func (*schemaResolver) UpdateUser(ctx context.Context, args *struct {
 		DisplayName: args.DisplayName,
 		AvatarURL:   args.AvatarURL,
 	}
-	if args.Username != nil {
+	if args.Username != nil && viewerCanChangeUsername(ctx, userID) {
```

If they attempt to change the username and are unable to, this should return an error (`unable to change username because auth.disableUsernameChanges is true in critical config`) instead of silently discarding the update.